python/cugraph/core/core_number.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from cugraph.core import core_number_wrapper
+
+
+def core_number(G):
+    """
+    Compute the core numbers for the nodes of the graph G.
+
+    Parameters
+    ----------
+    graph : cuGraph.Graph
+        cuGraph graph descriptor with connectivity information. The graph can
+        contain either directed or undirected edges where undirected edges are
+        represented as directed edges in both directions.
+
```

Comment:
1. That's pretty obvious but it doesn't harm to clarify that weights are not used in this algo.
2. I read in Nx that self-loops are not supported in their implementation, is this ok in yours? If not we should say it here.

Reply:
1. Fixed
2. NetworkX does not support self loops and parallel edges. Hornet does not support this as well. However, the difference is that our implementation does not check for this while NetworkX does before running core_number/kcore. Should there be a function implemented that checks for this error?

Reply:
It would be useful to have a helper function for checking self-loops in cuGraph. Once it is available we can add this to k-core pipelines and others. I don't want to block this PR for that though. https://github.com/rapidsai/cugraph/issues/536

Reply:
What really matters for this PR is that not supporting self loops is clearly mentioned in the documentation.